pkg/kubelet/dockershim/helpers.go

```diff
@@ -382,8 +384,34 @@ func ensureSandboxImageExists(client libdocker.Interface, image string) error {
 	if !libdocker.IsImageNotFoundError(err) {
 		return fmt.Errorf("failed to inspect sandbox image %q: %v", image, err)
 	}
-	err = client.PullImage(image, dockertypes.AuthConfig{}, dockertypes.ImagePullOptions{})
+
+	// To support images in private registries, try to read docker config
+	authConfig := dockertypes.AuthConfig{}
+	keyring := &credentialprovider.BasicDockerKeyring{}
+	pwd, _ := os.Getwd()
+	paths := []string{filepath.Join(pwd, ".docker"), filepath.Join(os.Getenv("HOME"), ".docker"), "/.docker"}
+	var cfgLoadErr error
+	if cfg, err := credentialprovider.ReadDockerConfigJSONFile(paths); err == nil {
+		keyring.Add(cfg)
+	} else if cfg, err := credentialprovider.ReadDockercfgFile(paths); err == nil {
+		keyring.Add(cfg)
+	} else {
+		cfgLoadErr = err
+	}
+	if creds, withCredentials := keyring.Lookup(image); withCredentials {
+		// Use the first one that matched our image
+		for _, cred := range creds {
+			authConfig.Username = cred.Username
+			authConfig.Password = cred.Password
+			break
+		}
+	}
+
+	err = client.PullImage(image, authConfig, dockertypes.ImagePullOptions{})
 	if err != nil {
+		if cfgLoadErr != nil {
+			glog.Infof("Docker config loading error: %v. If sandbox image %q is in a private registry, this will be a problem.", cfgLoadErr, image)
```

Comment:
Suggest `"Couldn't load Docker config. If sandbox image %q is in a private registry, this will cause further errors. Error: %v"`, as you never know how long the error message will be, putting it at the end often makes it easier to read logs.

Consider logging as `glog.Errorf`, since you technically encountered an error.